dts/arm/yaml/atmel,sam-uart.yaml

```diff
@@ -0,0 +1,30 @@
+---
+title: SAM Family UART
+id: atmel,sam-uart
+version: 0.1
+
+description: >
+    This binding gives a base representation of the SAM UART
+
+inherits:
+  - !include uart.yaml
+
+properties:
+  - compatible:
+      type: string
+      category: required
+      description: compatible strings
+      constraint: "atmel,sam-uart"
+
+  - reg:
```

Comment:
The main point of introducing DTS in Linux was to enable running the same binary on different hardware. In this case register addresses cannot be hard coded in the main image as they will vary between platforms. So DTS neatly separates driver logic from hardware data. But this is not the case in Zephyr. I haven't heard about any plans for Zephyr to support running the same image on different hardware. That's also quite unlikely as we want the image to be the smallest possible. In case of Zephyr we use DTS merely as a convenient way to configure the image.

For Atmel SoCs reg, interrupts and clocks properties are not required as all these data is provided by ASF header files.